vulnerability/fetcher_test.go

```diff
@@ -35,14 +35,31 @@ import (
 	"github.com/elastic/cloudbeat/resources/utils/testhelper"
 )
 
-func TestVulnerabilityFetcher_FetchInstances(t *testing.T) {
+type FetcherTestSuite struct {
+	suite.Suite
+	opts goleak.Option
+}
+
+func TestFetcherTestSuite(t *testing.T) {
+	s := new(FetcherTestSuite)
+	s.opts = goleak.IgnoreCurrent()
+	suite.Run(t, s)
+}
+
+func (s *FetcherTestSuite) TearDownTest() {
+	// Verify no goroutines are leaking. Safest to keep this on top of the function.
+	// Go defers are implemented as a LIFO stack. This should be the last one to run.
+	goleak.VerifyNone(s.T(), s.opts)
```

Comment:
I don't mind the approach but FYI you don't need a full test suite, you just need to `defer` the call from the test function.

Reply:
I know, I just aligned with other tests we have already that call `VerifyNone`